drivers/npmx/npmx_shell.c

```diff
@@ -2706,6 +2727,289 @@ static int cmd_vbusin_status_connected_get(const struct shell *shell, size_t arg
+
+	return 0;
+}
+
+static int cmd_ship_config_time_get(const struct shell *shell, size_t argc, char **argv)
+{
+	ARG_UNUSED(argc);
+	ARG_UNUSED(argv);
+
+	return ship_config_get(shell, SHIP_CONFIG_TYPE_TIME);
+}
+
+static int cmd_ship_config_time_set(const struct shell *shell, size_t argc, char **argv)
+{
+	return ship_config_set(shell, argc, argv, SHIP_CONFIG_TYPE_TIME);
+}
+
+static int cmd_ship_config_inv_polarity_get(const struct shell *shell, size_t argc, char **argv)
+{
+	ARG_UNUSED(argc);
+	ARG_UNUSED(argv);
+
+	return ship_config_get(shell, SHIP_CONFIG_TYPE_INV_POLARITY);
+}
+
+static int cmd_ship_config_inv_polarity_set(const struct shell *shell, size_t argc, char **argv)
+{
+	return ship_config_set(shell, argc, argv, SHIP_CONFIG_TYPE_INV_POLARITY);
+}
+
+static int ship_reset_config_get(const struct shell *shell, ship_reset_config_type_t config_type)
+{
+	npmx_instance_t *npmx_instance = npmx_driver_instance_get(pmic_dev);
+
+	if (npmx_instance == NULL) {
+		shell_error(shell, "Error: shell is not initialized.");
+		return 0;
+	}
+
+	npmx_ship_reset_config_t reset_config;
+	npmx_ship_t *ship_instance = npmx_ship_get(npmx_instance, 0);
+	npmx_error_t err_code = npmx_ship_reset_config_get(ship_instance, &reset_config);
+
+	if (check_error_code(shell, err_code)) {
+		bool config_val = false;
+
+		switch (config_type) {
+		case SHIP_RESET_CONFIG_TYPE_LONG_PRESS:
+			config_val = reset_config.long_press;
+			break;
+
+		case SHIP_RESET_CONFIG_TYPE_TWO_BUTTONS:
+			config_val = reset_config.two_buttons;
+			break;
+		}
+
+		shell_print(shell, "Value: %d.", (int)config_val);
+	} else {
+		shell_error(shell, "Error: unable to read reset config.");
+	}
+
+	return 0;
+}
+
+static int ship_reset_config_set(const struct shell *shell, size_t argc, char **argv,
+				 ship_reset_config_type_t config_type)
+{
+	npmx_instance_t *npmx_instance = npmx_driver_instance_get(pmic_dev);
+
+	if (npmx_instance == NULL) {
+		shell_error(shell, "Error: shell is not initialized.");
+		return 0;
+	}
+
+	if (argc < 2) {
+		shell_error(shell, "Error: missing config value.");
+		return 0;
+	}
+
+	int err = 0;
+	uint32_t config_val = shell_strtoul(argv[1], 0, &err);
+
+	if (err != 0) {
+		shell_error(shell, "Error: config value must be an integer.");
+		return 0;
+	}
+
+	npmx_ship_reset_config_t reset_config;
+	npmx_ship_t *ship_instance = npmx_ship_get(npmx_instance, 0);
+
+	npmx_error_t err_code = npmx_ship_reset_config_get(ship_instance, &reset_config);
+
+	if (!check_error_code(shell, err_code)) {
+		shell_error(shell, "Error: unable to read reset config.");
+		return 0;
+	}
+
+	switch (config_type) {
+	case SHIP_RESET_CONFIG_TYPE_LONG_PRESS:
+		reset_config.long_press = !!config_val;
+		break;
+	case SHIP_RESET_CONFIG_TYPE_TWO_BUTTONS:
+		reset_config.two_buttons = !!config_val;
+		break;
+	}
+
+	err_code = npmx_ship_reset_config_set(ship_instance, &reset_config);
+
+	if (check_error_code(shell, err_code)) {
+		shell_print(shell, "Success: %d.", (int)config_val);
```

Comment:
I don't like the fact that this is not the same as value that was set (`!!config_val`).
It would be best to cast it to bool in both cases.